conn_go18.go

```diff
@@ -49,41 +51,57 @@ func (cn *conn) BeginTx(ctx context.Context, opts driver.TxOptions) (driver.Tx,
 	if err != nil {
 		return nil, err
 	}
-	cn.txnClosed = cn.watchCancel(ctx)
+	cn.txnFinish = cn.watchCancel(ctx)
 	return tx, nil
 }
 
-func (cn *conn) watchCancel(ctx context.Context) chan<- struct{} {
+func (cn *conn) watchCancel(ctx context.Context) func() {
 	if done := ctx.Done(); done != nil {
-		closed := make(chan struct{})
+		finished := make(chan struct{})
 		go func() {
 			select {
 			case <-done:
-				cn.cancel()
-			case <-closed:
+				_ = cn.cancel()
+				finished <- struct{}{}
+			case <-finished:
 			}
 		}()
-		return closed
+		return func() {
+			select {
+			case <-finished:
+			case finished <- struct{}{}:
+			}
+		}
 	}
 	return nil
 }
 
-func (cn *conn) cancel() {
-	var err error
-	can := &conn{}
-	can.c, err = dial(cn.dialer, cn.opts)
+func (cn *conn) cancel() error {
```

Comment:
Why are we returning errors here at all? They are never used. In case we want them in the future and it's just good practice?